### PR TITLE
add docker image for gpctl and kubecdl

### PR DESCRIPTION
**Pull request by sagor999**
## Description

add docker image for gpctl and kubecdl so that we don't need to build it in other werft jobs

## Related Issue(s)

Fixes https://github.com/gitpod-io/ops/issues/733

## How to test

```
curl -L https://github.com/csweichel/oci-tool/releases/download/v0.2.0/oci-tool_0.2.0_linux_amd64.tar.gz | tar xz
chmod +x oci-tool

./oci-tool fetch file eu.gcr.io/gitpod-core-dev/dev/dev-utils:commit-d66b864f634482937265b3b7bfb1ed09df731bd7 app/gpctl
chmod +x gpctl

./gpctl --help
```

## Release Notes

```release-note
Add docker images for gpctl and kubecdl
```

### CODE REVIEW

**Comment by roboquat** [comment]
@sagor999: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by princerachit** [comment]
I like this, but on the same time I can recall some old discussion on why we don't add gpctl as build target (my memory is not serving me well). I will wait for @csweichel approval before merging this PR.

/hold

**Review by princerachit — APPROVED** [review]

**Comment by csweichel** [comment]
I've just added a new command to the oci-tool ([v0.2.0 release](https://github.com/csweichel/oci-tool/releases/tag/v0.2.0)) which lets us download files from an image layer, e.g. the installer:
```
oci-tool fetch file eu.gcr.io/gitpod-core-dev/build/installer:main.2583 app/installer
```

The same would work for `gpctl` or `kubecdl`.

**Comment by sagor999** [comment]
@csweichel @princerachit PTAL

**Comment by utam0k** [comment]
@sagor999 I couldn't see the previous commit 😭 Maybe do we have to update the description of PR?

**Comment by sagor999** [comment]
@utam0k thank you, you are right. I updated how to test section.

**Review by utam0k — APPROVED** [review]
The code itself worked fine.

File: dev/BUILD.yaml
Comment (csweichel) [inline dev/BUILD.yaml:30]: nit: this package is unlikely to be deployed to a cluster, hence doesn't need this entry

**Review by csweichel — DISMISSED** [review]

File: dev/BUILD.yaml
Comment (sagor999) [reply dev/BUILD.yaml]: Ah, thank you! I was not quite sure how that value is used so I just copy pasted it from another file :) 
Fixed!

**Review by kylos101 — APPROVED** [review]
Works well! Thanks @sagor999 and @csweichel for the `oci-tool` feature add. Also, good to know about `scratch`.

I downloaded like so `./oci-tool fetch file eu.gcr.io/gitpod-core-dev/dev/dev-utils:pavel-gpctl.10 app/gpctl`.

**Comment by kylos101** [comment]
/unhold

Feedback from @csweichel has been addressed, hence removing hold.

**Pull request merged**
